app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ActiveRecord::Base
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_user_posts_counter
```

Comment:
- [ OPTIONAL ] I will suggest that you consider using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) : `after_save` to call the `#update_user_posts_counter` method every time a new post has been saved to database, then make the `update_user_posts_counter` method private. 

- Example Implementation
```ruby
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end

```